File: core/evidence/unsupported_expansion_engine.py
from __future__ import annotations

from typing import Any, Dict, List


def detect_unsupported_expansion(
    evidence: List[str],
    expansion_type: str,
    count: int,
) -> Dict[str, Any]:
    suppressed = count > 0 and len(evidence) < 2
    return {
        "expansion_type": expansion_type,
        "count": count,
        "suppressed": suppressed,
        "unsupported_expansions": [f"{expansion_type}:{i}" for i in range(count)] if suppressed else [],
        "reason": "insufficient_evidence" if suppressed else "allowed",
    }

File: src/evidence/unsupportedExpansionEngine.ts
/**
 * Converted from Python: core/evidence/unsupported_expansion_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function detectUnsupportedExpansion(evidence: any, expansion_type: any, count: any): any {
  var suppressed: any = py.and2((count > 0), () => ((py.len(evidence) < 2)));
  return {"expansion_type": expansion_type, "count": count, "suppressed": suppressed, "unsupported_expansions": (py.truthy(suppressed) ? py.range(count).map((i: any) => `${py.toStr(expansion_type)}:${py.toStr(i)}`) : []), "reason": (py.truthy(suppressed) ? "insufficient_evidence" : "allowed")};
}
